generator/internal/dart/annotate_test.go

```diff
@@ -247,3 +247,171 @@ func TestCalculateRequiredFields(t *testing.T) {
 		t.Errorf("mismatch in TestCalculateRequiredFields (-want, +got)\n:%s", diff)
 	}
 }
+
+func TestBuildQueryLines(t *testing.T) {
+	for _, test := range []struct {
+		field *api.Field
+		want  []string
+	}{
+		// primitives
+		{
+			&api.Field{Name: "bool", JSONName: "bool", Typez: api.BOOL_TYPE},
+			[]string{"if (result.bool != null) 'bool': '${result.bool}'"},
+		}, {
+			&api.Field{Name: "int32", JSONName: "int32", Typez: api.INT32_TYPE},
+			[]string{"if (result.int32 != null) 'int32': '${result.int32}'"},
+		}, {
+			&api.Field{Name: "int64", JSONName: "int64", Typez: api.INT64_TYPE},
+			[]string{"if (result.int64 != null) 'int64': '${result.int64}'"},
+		}, {
+			&api.Field{Name: "double", JSONName: "double", Typez: api.DOUBLE_TYPE},
+			[]string{"if (result.double != null) 'double': '${result.double}'"},
+		}, {
+			&api.Field{Name: "string", JSONName: "string", Typez: api.STRING_TYPE},
+			[]string{"if (result.string != null) 'string': result.string!"},
+		},
+
+		// repeated primitives
+		{
+			&api.Field{Name: "boolList", JSONName: "boolList", Typez: api.BOOL_TYPE, Repeated: true},
+			[]string{"if (result.boolList != null) 'boolList': result.boolList!.map((e) => '$e')"},
+		}, {
+			&api.Field{Name: "int32List", JSONName: "int32List", Typez: api.INT32_TYPE, Repeated: true},
+			[]string{"if (result.int32List != null) 'int32List': result.int32List!.map((e) => '$e')"},
+		}, {
+			&api.Field{Name: "int64List", JSONName: "int64List", Typez: api.INT64_TYPE, Repeated: true},
+			[]string{"if (result.int64List != null) 'int64List': result.int64List!.map((e) => '$e')"},
+		}, {
+			&api.Field{Name: "doubleList", JSONName: "doubleList", Typez: api.DOUBLE_TYPE, Repeated: true},
+			[]string{"if (result.doubleList != null) 'doubleList': result.doubleList!.map((e) => '$e')"},
+		}, {
```

Comment:
Oh, that is much nicer than putting the braces on separate lines.  I am going to steal that idea.